setup.py

```diff
@@ -0,0 +1,16 @@
+import os
```

Comment:
Do we really need this? It's not on pypi and I don't think we have any intention of putting it there nor is it needed for releasing to prod or locally running in dev.

Reply:
Yep. 

* This is what allows `pip install -e .` to work in `requirements-dev.in`
* ... making `py_proxy` available in the `tox` environment where `pserve` is located
* ... allowing `pserve` to find the class mentioned in the `conf/development.ini` lines:

```
[app:main]
use = call:py_proxy.app:app
```
*phew!*

To see this work (or not) have a go deleting the file and running `make dev`. It should fail with complaints about not being able to find `py_proxy`, unless its cached somehow.

Reply:
Seems to be one of the things our apps are inconsistent about. LMS has a `setup.py`. So does Potemkin. h and bouncer somehow manage without one. The cookiecutters that come with Pyramid (https://github.com/Pylons/pyramid-cookiecutter-starter and https://github.com/Pylons/pyramid-cookiecutter-alchemy) also create one.

Reply:
I suspect that h and bouncer not having a `setup.py` is another case of "we know about than Pyramid" or "we know better than the normal way of doing things" which kind of plagued Hypothesis for a while and has left quite some mess behind. I'm sure there _was_ a reason for it but can't remember what

Reply:
h's `setup.py` was [removed in 2017](https://github.com/hypothesis/h/pull/4514) but it doesn't say why. Bouncer's was also [removed in 2017](https://github.com/hypothesis/bouncer/pull/36) as part of "converting bouncer to an application" because "it doesn't really make sense to keep bouncer as a Python package anymore in my opinion." One of the popular blog posts out there about Python packages vs Python applications will explain a way of doing apps without `setup.py` and why, I'm sure.

There seems to have been a general attitude that apps like bouncer and h shouldn't be installable Python packages because they aren't intended for other people to reuse them, aren't intended for redistribution of for uploading to PyPI (various brief comments to this effect in ancient Slack history if you search for "setup.py").

My opinion: Hypothesis isn't special enough to deviate from the normal way that Pyramid does things, especially on basics like this that we should not be wasting our time on. `setup.py` is _not_ only used for installing the package and listing its dependencies, as past comments seem to claim. Having no `setup.py` breaks things like Pyramid's builtin `pshell` command and other builtin scripts, breaks the standard way of adding your own custom command line scripts to a Pyamid app, and who knows what else. While I'm sure that can all be reinvented it seems like a waste of time to do so.